File: eCommerce.Services/Database/ProductRecommendation.cs

using System.ComponentModel.DataAnnotations;

namespace eCommerce.Services.Database
{
    public class ProductRecommendation
    {
        [Key]
        public int Id { get; set; }

        public int ProductId { get; set; }
        public Product Product { get; set; } = null!;

        public int RecommendedProductId { get; set; }
        public Product RecommendedProduct { get; set; } = null!;

        [Required]
        public float Score { get; set; }
    }
}

File: UI/ecommerce_mobile/lib/models/product_recommendation.dart
import 'package:ecommerce_mobile/models/product.dart';
import 'package:json_annotation/json_annotation.dart';

part 'product_recommendation.g.dart';

@JsonSerializable()
class ProductRecommendation {
  final int? id;
  final int? productId;
  final Product? product;
  final int? recommendedProductId;
  final Product? recommendedProduct;
  final double? score;

  ProductRecommendation({
    this.id,
    this.productId,
    this.product,
    this.recommendedProductId,
    this.recommendedProduct,
    this.score
  });

  factory ProductRecommendation.fromJson(Map<String, dynamic> json) =>
      _$ProductRecommendationFromJson(json);

  Map<String, dynamic> toJson() => _$ProductRecommendationToJson(this);
}